server.go

```diff
@@ -203,47 +203,79 @@ func newServer(listenAddrs []string, chanDB *channeldb.DB, cc *chainControl,
 		selfAddrs = append(selfAddrs, lnAddr)
 	}
 
-	chanGraph := chanDB.ChannelGraph()
-
 	// TODO(roasbeef): make alias configurable
 	alias, err := lnwire.NewNodeAlias(hex.EncodeToString(serializedPubKey[:10]))
 	if err != nil {
 		return nil, err
 	}
-	selfNode := &channeldb.LightningNode{
-		HaveNodeAnnouncement: true,
-		LastUpdate:           time.Now(),
-		Addresses:            selfAddrs,
-		PubKey:               privKey.PubKey(),
-		Alias:                alias.String(),
-		Features:             globalFeatures,
-	}
 
-	// If our information has changed since our last boot, then we'll
-	// re-sign our node announcement so a fresh authenticated version of it
-	// can be propagated throughout the network upon startup.
-	//
-	// TODO(roasbeef): don't always set timestamp above to _now.
+	timeStamp := time.Now()
 	nodeAnn := &lnwire.NodeAnnouncement{
-		Timestamp: uint32(selfNode.LastUpdate.Unix()),
-		Addresses: selfNode.Addresses,
-		NodeID:    selfNode.PubKey,
+		Timestamp: uint32(timeStamp.Unix()),
+		Addresses: selfAddrs,
+		NodeID:    privKey.PubKey(),
 		Alias:     alias,
-		Features:  selfNode.Features,
+		Features:  globalFeatures,
 	}
-	selfNode.AuthSig, err = discovery.SignAnnouncement(s.nodeSigner,
-		s.identityPriv.PubKey(), nodeAnn,
-	)
-	if err != nil {
-		return nil, fmt.Errorf("unable to generate signature for "+
-			"self node announcement: %v", err)
+
+	updateNodeAnn := true
+
+	chanGraph := chanDB.ChannelGraph()
```

Comment:
I would move these two declarations to right before where they are used the first time